### PR TITLE
Fixed #171 - Fam_backup regression test crashes 

### LINKED ISSUE #171: Fam backup regression test crashes
Following crash was seen with fam backup restore test case in CIS. 
The crash is seen centos7 scaleup system.

We will disable this test case for now. 

(gdb) bt
#0 0x00007fad14f7956d in ?? () from /lib64/libstdc++.so.6
#1 0x0000000000567c5e in ~basic_string (this=0x7facf80067c8, __in_chrg=<optimized out>) at /usr/include/c++/4.8.2/bits/basic_string.h:539
#2 openfam::Fam_Backup_Tag::~Fam_Backup_Tag (this=0x7facf8006780, __in_chrg=<optimized out>)
at /home/abdulla/centos7_devel/OpenFAM/src/common/fam_async_qhandler.h:87
#3 0x000000000056b758 in openfam::Fam_CIS_Async_Handler::CallData::Proceed (this=0x7facf80008c0, type=type@entry=openfam::RT_BACKUP)
at /home/abdulla/centos7_devel/OpenFAM/src/cis/fam_cis_async_handler.h:237
#4 0x000000000056ca94 in openfam::Fam_CIS_Async_Handler::HandleRpcs<Fam_CIS_Rpc::WithAsyncMethod_delete_backup<Fam_CIS_Rpc::WithAsyncMethod_restore<Fam_CIS_Rpc::WithAsyncMethod_backup<Fam_CIS_Rpc::WithAsyncMethod_copy<openfam::Fam_CIS_Server> > > > > (this=<optimized out>, service=<optimized out>, cq=0xdd12b0,
famCIS=<optimized out>, type=openfam::RT_BACKUP) at /home/abdulla/centos7_devel/OpenFAM/src/cis/fam_cis_async_handler.h:414
#5 0x00007fad14fd3330 in ?? () from /lib64/libstdc++.so.6
#6 0x00007fad147f1ea5 in start_thread () from /lib64/libpthread.so.0
#7 0x00007fad1451ab0d in clone () from /lib64/libc.so.6

### PR DESCRIPTION
**Pull request by BA-Ramya**
- Fixed compilation issue with pmi2 unit test case.

### CODE REVIEW

**Review by mashoodk — APPROVED** [review]

**Review by faizan-barmawer — CHANGES_REQUESTED** [review]
We write the test scenario such that we know whether the API is expected to succeed or throw an exception. It is better to use appropriate gtest  EXPECT_* function to report the success/failure of the test scenario. 
Any reg-test failure indicates that either there is a problem with test scenairo, test system configuration or there is implementaion issue. 

Regards,
Faizan

File: src/allocator/memserver_allocator.cpp [hunk 1]
```diff
@@ -584,24 +585,30 @@ Fam_Backup_Info Memserver_Allocator::get_backup_info(const string BackupName,
...
-            info.gid = std::stoi(value);
+            if (attribute.compare("gid") == 0)
+                info.gid = std::stoi(value);
 
-        if (attribute.compare("mode") == 0)
-            info.mode = std::stoi(value);
+            if (attribute.compare("mode") == 0)
+                info.mode = std::stoi(value);
+        } catch (...) {
+            THROW_ERRNO_MSG(Memory_Service_Exception, BACKUP_DATA_INVALID,
```
[the remark this answers is not in the document]
Comment (BA-Ramya) [reply src/allocator/memserver_allocator.cpp]: Addressed

File: test/reg-test/fam-api-reg/fam_backup_restore_reg_test.cpp [hunk 2]
```diff
@@ -64,21 +64,43 @@ TEST(FamBackupRestore, BackupSuccess) {
...
+    Fam_Backup_Options *backupOptions =
+        (Fam_Backup_Options *)malloc(sizeof(Fam_Backup_Options));
+
+    char *backupName = (char *)malloc(FILE_MAX_LEN);
+    try {
+        sprintf(backupName, "%s.%s", testRegion, firstItem);
+        void *waitobj1 = my_fam->fam_backup(item, backupName, backupOptions);
+        my_fam->fam_backup_wait(waitobj1);
+    } catch (Fam_Exception &e) {
+        cout << "fam_backup exception: " << e.fam_error_msg() << endl;
```
[the remark this answers is not in the document]
Comment (BA-Ramya) [reply test/reg-test/fam-api-reg/fam_backup_restore_reg_test.cpp]: For some scenarios, the exception can happen either during fam_backup/fam_backup_wait depending on the way services are configured. For such cases, a flag is being used and set if exceptions occur. This flag value is verified after try catch block to ensure that exception is raised. For rest of scenarios EXPECT_THROW is being used.

**Review by mashoodk — APPROVED** [review]

File: src/common/fam_internal_exception.h [hunk 3]
```diff
@@ -188,6 +193,7 @@ inline enum Fam_Error convert_to_famerror(enum Internal_Error serverErr) {
     case ITEM_REGISTRATION_FAILED:
     case ITEM_DEREGISTRATION_FAILED:
     case DATAITEM_SIZE_TOO_LONG:
+    case BACKUP_SIZE_TOO_LONG:
```
Comment (faizan-barmawer) [inline src/common/fam_internal_exception.h:196]: Should we use "LARGE" in place of "LONG" while using it with data item/backup "size" ?
minor comment, you can ignore this.

File: test/reg-test/fam-api-reg/fam_min_max_atomics_mt_reg_test.cpp [hunk 4]
```diff
@@ -62,7 +62,7 @@ int rc;
 #define REGION_SIZE (32 * 1024 * NUM_THREADS)
 #define REGION_PERM 0777
 
-#define SHM_CHECK (strcmp(openFamModel, "shared_memory") == 0)
+//#define SHM_CHECK (strcmp(openFamModel, "shared_memory") == 0)
```
Comment (faizan-barmawer) [inline test/reg-test/fam-api-reg/fam_min_max_atomics_mt_reg_test.cpp:65]: Do you want to leave this line commented or want to delete?

**Review by faizan-barmawer — APPROVED** [review]
Code changes look good.
I have few minor comments in the code, that can be ignored.

Regards,
Faizan

**Review by mashoodk — APPROVED** [review]

**Pull request merged**